src/django_upgrade/fixers/email_apis.py

```diff
@@ -0,0 +1,126 @@
+"""
+Convert positional arguments to keyword arguments for Django email APIs:
+https://docs.djangoproject.com/en/6.0/releases/6.0/#positional-arguments-in-django-core-mail-apis
+"""
+
+from __future__ import annotations
+
+import ast
+from collections.abc import Iterable
+from functools import partial
+
+from tokenize_rt import Offset, Token
+
+from django_upgrade.ast import ast_start_offset
+from django_upgrade.data import Fixer, State, TokenFunc
+from django_upgrade.tokens import CODE, OP, find, parse_call_args
+
+fixer = Fixer(
+    __name__,
+    min_version=(6, 0),
+)
+
+MODULE = "django.core.mail"
+
+# Map function names to their allowed positional parameters count and keyword-only parameter names
+EMAIL_FUNCTION_CONFIG = {
+    "send_mail": {
+        "max_positional": 4,  # subject, message, from_email, recipient_list
+        "keyword_params": ["fail_silently", "auth_user", "auth_password", "connection", "html_message"]
+    },
+    "send_mass_mail": {
+        "max_positional": 1,  # datatuple
+        "keyword_params": ["fail_silently", "auth_user", "auth_password", "connection"]
+    },
+    "mail_admins": {
+        "max_positional": 2,  # subject, message
+        "keyword_params": ["fail_silently", "connection", "html_message"]
+    },
+    "mail_managers": {
+        "max_positional": 2,  # subject, message
+        "keyword_params": ["fail_silently", "connection", "html_message"]
+    },
+}
+
+
+@fixer.register(ast.Call)
+def visit_Call(
+    state: State,
+    node: ast.Call,
+    parents: tuple[ast.AST, ...],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    func_name = None
+    
+    # Check for direct import: from django.core.mail import send_mail
+    if (
+        isinstance(node.func, ast.Name)
+        and node.func.id in EMAIL_FUNCTION_CONFIG
+        and node.func.id in state.from_imports[MODULE]
+    ):
+        func_name = node.func.id
+    # Check for module import: from django.core import mail; mail.send_mail(...)
+    elif (
+        isinstance(node.func, ast.Attribute)
+        and node.func.attr in EMAIL_FUNCTION_CONFIG
+        and isinstance(node.func.value, ast.Name)
+        and node.func.value.id == "mail"
+        and "mail" in state.from_imports["django.core"]
+    ):
+        func_name = node.func.attr
+    
+    if func_name is not None:
```

Comment:
Turn this into one large conditional, to avoid the overhead of creating a variable for every `ast.Call` in the tree. use walruse poerator in conditional to bind `func_name`.